### PR TITLE
feat(web): simulate transactions from queue

### PR DESCRIPTION
**Pull request by schmanu**
## How this PR fixes it
New button in queued transactions' details to simulate the transaction.

New component `QueuedTxSimulation` which offers simulation without requiring a Transaction in the context and renders it inline.

## How to test it
1. Queue any transaction
2. Simulate it in the tx details
3. Test edge cases 
  - by queueing txs that will fail (by using e.g. tx builder)
  - by blocking the simulation request in the browser dev tools

## Screenshots
Screenshot 2025-05-30 at 16 55 24
Screenshot 2025-05-30 at 16 55 49
Screenshot 2025-05-30 at 16 55 32

## Checklist

- [x] I've tested the branch on mobile 📱
- [x] I've documented how it affects the analytics (if at all) 📊
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

### CODE REVIEW

File: apps/web/src/components/tx/security/tenderly/utils.ts [hunk 1]
```diff
@@ -106,3 +109,41 @@ export const getSimulationPayload = async (params: SimulationTxParams): Promise<
...
+
+export type SimulationStatus = {
+  isLoading: boolean
+  isFinished: boolean
+  isSuccess: boolean
+  isCallTraceError: boolean
+  isError: boolean
+}
+
+export const getSimulationStatus = (simulation: UseSimulationReturn): SimulationStatus => {
```
Comment (compojoom) [inline apps/web/src/components/tx/security/tenderly/utils.ts]: Can't this live in the utils package? It looks like something that we would also need in the mobile app.

Comment (schmanu) [reply apps/web/src/components/tx/security/tenderly/utils.ts]: There is no issue moving it there.
But should we move it there when the mobile app starts using it? Or should I do it now?

**Comment by usame-algan** [comment]
What is the expected behaviour if there is a transaction in the queue that I already signed with my wallet? In that case the simulation fails for me when trying to simulate again.

**Comment by schmanu** [comment]
> What is the expected behaviour if there is a transaction in the queue that I already signed with my wallet? In that case the simulation fails for me when trying to simulate again.

This should not happen. Seems like the overwrites are somehow wrong in that case.
I will take a look and fix it.

File: apps/web/src/components/tx/security/tenderly/utils.ts (hunk 1 above)
Comment (compojoom) [reply apps/web/src/components/tx/security/tenderly/utils.ts]: Let's move it now. If we have utility code that is not dependent on any mobile/web stuff it should live in the utils package. This makes it easier to use it later.

**Comment by schmanu** [comment]
@usame-algan 
It was a general bug in the simulation utils. It should be fixed now.
Good catch!

**Review by compojoom — APPROVED** [review]
